recipe/meta.yaml

```
@@ -239,7 +243,9 @@ outputs:
         - {{ pin_subpackage('libgomp', exact=True) }}
       run:
         - _libgcc_mutex * main
-        - {{ pin_subpackage("libgomp", max_pin="x") }}
+        - {{ pin_subpackage("libgomp", max_pin=None) }}
+      run_constrained:
+        - libgcc-ng * openmp_impl*
```

Why do we need the 2 lines above? Seems unnecessary

---

We cannot mix `openmp_impl=4.5=0_llvm` with any version of `libgcc-ng` because the older ones will not have the symlinks for `libgomp.so.1` moved to a separate package. This will cause clobber errors when the conda environments are built and could cause silent bugs where LLVM OpenMP is not actually being used at run time.

---

We could instead create another mutex package, but using build strings and run_constrained means we don't need to patch repo metadata.

---

I have added the `run_constrained` section to the other PR as well.

---

Okay, then let's do as below,
```suggestion
        - libgcc-ng {{ ctng_gcc }} *_{{ build_num | default(0) }}
```

and remove the `openmp_impl` build string.

---

We can remove older `libgcc-ng` builds later.

---

I'm not sure this will work. It is actually **any** previous build of libgcc-ng, even those on defaults. We cannot remove the defaults builds or control their build numbers etc to make sure the condition you suggested will always do what we intend it to do.

---

I'd rather not worry about defaults and rely on conda solver using the latest build.

---

I really don't think this will work and is a recipe for silent bugs. Are you sure?

---

What if we rename the libgcc-ng package to something else so the solver never mixes them?

---

That would definitely result in clobbering

---

How about changing `_libgcc_mutex * main` to `_libgcc_mutex * conda-forge` to ensure conda-forge builds are used?

---

Yep, we can change the mutex, but we only get to do that once unless we version the mutex build string too. 

Where is the mutext package made? I will need to put in a PR for that too.

---

It's at https://github.com/conda-forge/_libgcc_mutex-feedstock/blob/master/recipe/meta.yaml
Yes, we can change to an exact pinning of `_libgcc_mutex`